src/Symfony/Component/Translation/MessageSelector.php

```diff
@@ -49,10 +49,16 @@ class MessageSelector
      */
     public function choose($message, $number, $locale)
     {
-        preg_match_all('/(?:\|\||[^\|])++/', $message, $parts);
+        $parts = array();
+        if (preg_match('/^\|+$/', $message)) {
```

Comment:
please change the regex to `^\|++$` to avoid useless backtracking (which could cause issues with long strings). Backtracking will never allow to find a match in a different way for this regex.